pkl/private/pkl.bzl

```
@@ -231,6 +232,15 @@ def _pkl_eval_impl(ctx):
     )
     return [DefaultInfo(files = depset(outputs), runfiles = ctx.runfiles(outputs))]
 
+def _find_common_prefix(strings):
+    reference = strings[0]
+
+    for i in range(len(reference)):
+        if any([s[i] != reference[i] for s in strings if i < len(s)]):
+            return reference[:i]
+
+    return reference
```

You'll need to split the path into an array of components (otherwise you'll find that the common prefix of [`somedir/two`, `somedir/three`] is `somedir/t`).

```suggestion
def _find_common_prefix(strings):
    paths = [s.split("/") for s in strings]
    reference = paths[0]

    for i in range(len(reference)):
        if any([s[i] != reference[i] for s in paths if i < len(s)]):
            return "/".join(reference[:i])

    return reference
```